certbot-dns-dyn/certbot_dns_dyn/dns_dyn_test.py

```diff
@@ -0,0 +1,218 @@
+"""Tests for certbot_dns_dyn.dns_dyn."""
+
+import os
+
+import unittest
+import mock
+
+from certbot import errors
+from certbot.plugins import dns_test_common
+from certbot.tests import util as test_util
+
+from certbot_dns_dyn.dns_dyn import Authenticator
+from certbot_dns_dyn.dns_dyn import rrem
+
+from dyn.tm.errors import DynectAuthError
+from dyn.tm.errors import DynectCreateError
+from dyn.tm.errors import DynectDeleteError
+
+DYN_CUSTOMER = 'test-customer'
+DYN_USERNAME = 'test-username'
+DYN_PASSWORD = 'test-password'
+
+class AuthenticatorTest(test_util.TempDirTestCase,
+                        dns_test_common.BaseAuthenticatorTest):
+    # pylint: disable=protected-access
+
+    def setUp(self):
+        super(AuthenticatorTest, self).setUp()
+
+        path = os.path.join(self.tempdir, 'dyn-test.ini')
+        dns_test_common.write({
+                                "dyn_customer": DYN_CUSTOMER,
+                                "dyn_username": DYN_USERNAME,
+                                "dyn_password": DYN_PASSWORD
+                              }, path)
+
+        self.config = mock.MagicMock(dyn_credentials=path,
+                                     dyn_propagation_seconds=0)
+        self.auth = Authenticator(self.config, "dyn")
+        self.auth._setup_credentials()
+
+    def test_rrem(self):
+        assert rrem('a.b.c', '.c') == 'a.b'
+        assert rrem('a.b.c', '.d') == 'a.b.c'
+
+    @mock.patch('certbot_dns_dyn.dns_dyn.DynectSession')
+    def test_get_dyn_client(self, patch_DynectSession):
+        self.auth._get_dyn_client()
+        patch_DynectSession.assert_called_once_with(DYN_CUSTOMER,
+                                                    DYN_USERNAME,
+                                                    DYN_PASSWORD)
+
+    @mock.patch('certbot_dns_dyn.dns_dyn.get_all_zones')
+    def test_find_zone_missing(self, patch_get_all_zones):
+        patch_get_all_zones.return_value = []
+        assert self.auth._find_zone('_acme-challenge.letsencrypt.org') == None
```

Comment:
I think it'd be wise to use `example.com` domain throughout the tests in general instead.